### DISCUSSION TITLE
How to Pass the value to a variable in terminal

### DISCUSSION THREAD

**Discussion by abdurraheemj** [seleniumbase/SeleniumBase#1204] [Q&A] [answered]
Hi Michael,
i have the script to decide if the script should choose the qa environment or production environment. I am going to move the script to azure devops. I need to avoid getting the input from the user instead I wanna pass the value(example:qa) to a variable via terminal. How to i achieve this?
env=""
qa="https://www.qa."
prd="https://www.prd"
 def setUp(self):
  super().setUp()
        
        self.env=input("Type qa or prd then press Enter:")
      
        
        if self.env=="qa":
            self.open(self.qa+self.homepage)
        elif self.env=="prd":
            self.open(self.prd+self.homepage)

**Comment by mdmintz** [comment, marked as answer]
Hi @abdurraheemj You can set the environment via the command-line using built-in environments that map to ``self.env``:

```bash
pytest --env=qa
pytest --env=production
```

In tests, access with ``self.env``. You'll have to type out "production" when using that one.
Then you can use that within ``if`` statements.

**Reply by abdurraheemj** [reply]
Thanks Michael, will try and reach you out for further help..

**Reply by abdurraheemj** [reply]
i'm not getting it. anyways thanks michael

**Reply by abdurraheemj** [reply]
if you can provide an example script, that would be great. god bless you

**Reply by mdmintz** [reply]
@abdurraheemj You already had the right idea with:

```python
if self.env == "qa":
    self.open(QA_URL)
elif self.env == "production":
    self.open(PRODUCTION_URL)
```

Those parameters get set on the command-line with:

```bash
pytest --env=qa
pytest --env=production
```

**Reply by Robert-Ho** [reply]
Hi Michael, I'm a newbee
how can use it in this case:
```
@step("Open the Swag Labs Login Page")
def go_to_swag_labs(context):
    sb = context.sb
    sb.open("https://www.saucedemo.com")
    sb.clear_local_storage()
```
Many thanks!

**Reply by mdmintz** [reply]
@Robert-Ho For the ``sb`` pytest fixture format, use ``sb.env`` instead of ``self.env``.

**Reply by Robert-Ho** [reply]
Hi, @mdmintz Thanks for your reply.

I have tried with it, but always got:
`sb.env = None`
![image](https://github.com/seleniumbase/SeleniumBase/assets/66453666/acd29ae3-4e88-44c4-9922-00e025a71c1e)


`behave login.feature -D env=qa`

```
@step("Open the Swag Labs Login Page")
def go_to_swag_labs(context):
    sb = context.sb
    if sb.env == "test":
        sb.open("https://www.saucedemo.com")
    else:
        print(f'sb.env = {sb.env}')
```
And one more think,
Can I get a demo for behave + pageObject.
Thank you!

**Reply by mdmintz** [reply]
@Robert-Ho See the example here: https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_sb_fixture.py for how to properly create tests that use the ``sb`` fixture. I have no idea what your ``context`` fixture is.

For ``behave`` BDD documentation, see the ReadMe here: https://github.com/seleniumbase/SeleniumBase/tree/master/examples/behave_bdd
And see the ``features/`` folder in there for example tests.

**Comment by mdmintz** [comment]
@abdurraheemj You already had the right idea with:

```python
if self.env == "qa":
    self.open(QA_URL)
elif self.env == "production":
    self.open(PRODUCTION_URL)
```

Those parameters get set on the command-line with:

```bash
pytest --env=qa
pytest --env=production
```

**Reply by abdurraheemj** [reply]
This is what i'm getting. 
ERROR: usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --env=qa
  inifile: None
  rootdir: /Users/user1/projects/automation/scripts/smoketest

**Reply by mdmintz** [reply]
Is seleniumbase installed? If you type "seleniumbase --version", you should see the following:

```bash
seleniumbase --version
 
   ______     __           _                 ____                
  / ____/__  / /__  ____  (_)_  ______ ___  / _  \____  ________ 
  \__ \/ _ \/ / _ \/ __ \/ / / / / __ `__ \/ /_) / __ \/ ___/ _ \
 ___/ /  __/ /  __/ / / / / /_/ / / / / / / /_) / (_/ /__  /  __/
/____/\___/_/\___/_/ /_/_/\__,_/_/ /_/ /_/_____/\__,_/____/\___/ 

...
seleniumbase 2.4.9
```

**Reply by abdurraheemj** [reply]
yes i do.all of my project in seleniumbase

**Reply by mdmintz** [reply]
@abdurraheemj When you run ``pytest --help``, do you see the SeleniumBase commands?

```bash
pytest --help

...

SeleniumBase command-line options for pytest:
  --browser={chrome,edge,firefox,ie,opera,phantomjs,safari,android,iphone,ipad,remote}
                        Specifies the web browser to use. Default: Chrome.
                        If you want to use Firefox, explicitly indicate that.
                        Example: (--browser=firefox)
  --chrome              Shortcut for --browser=chrome. On by default.)
  --edge                Shortcut for --browser=edge.)
  --firefox             Shortcut for --browser=firefox.)
  --ie                  Shortcut for --browser=ie.)
  --opera               Shortcut for --browser=opera.)
  --safari              Shortcut for --browser=safari.)
  --with-selenium       (DEPRECATED) Start tests with an open web browser.
                        (This is ALWAYS True now when importing BaseCase)
  --env={qa,staging,develop,production,master,remote,local,alpha,beta,main,test}
                        This option sets a test env from a list of choices.
 
...

```

**Reply by mdmintz** [reply]
@abdurraheemj I think I just lost your last message, but looks like you figured it out!

**Reply by abdurraheemj** [reply]
yes i see it. i really didn't think that variable and value  should be specific.
